File: Data Structures & Algorithms/maximum-product-subarray/submission-1.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        res = nums[0]
        n = len(nums)
        prefix = subfix = 0
        for i in range(len(nums)):
            prefix = nums[i] * (prefix or 1)
            subfix = nums[n -1 -i] * (subfix or 1)
            res = max(res,max(prefix , subfix))
        return res

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int i = 0;
        int res = 0;
        int maxf = 0;
        unordered_map<char,int> count;

        for ( int r = 0; r<s.size();r++){
            count[s[r]]++;
            maxf = max(maxf,count[s[r]]);
            while ((r-i+1)-maxf > k){
                count[s[i]]--;
                i++;
            }

            res = max(res,(r-i+1));
        }
        return res;
    }
};
